hazelcast/src/test/java/com/hazelcast/query/impl/predicates/NestedPredicateWithExtractorTest.java

```diff
@@ -124,6 +129,52 @@ public void nestedAttributeQuery_predicates() throws Exception {
     }
 
     @Test
+    public void nestedAttributeQuery_customPredicates_entryProcessor() throws Exception {
+        // GIVEN
+        map.put(1, new Body("body1", new Limb("hand")));
+        map.put(2, new Body("body2", new Limb("leg")));
+
+        // WHEN
+        Map<Integer, Object> result = map.executeOnEntries(new ExtractProcessor(), new CustomPredicate());
+        Body resultBody = (Body) Iterables.get(result.values(), 0);
```

Comment:
hmmm, Guava is on a (test) classpath only because it's a transitive dependency of a library for OSGi testing. 
I am not sure whether you want to (directly) depend on it.